src/tests/meson.build

```diff
@@ -0,0 +1,452 @@
+    link_with: [
+        libopensc,
+        libopensc_libraries
+    ],
+    dependencies: [
+        depcmocka,
+        libopensc_dependencies
+    ]
+)
+test('simpletlv', ut_simpletlv)
+
+ut_strip_pkcs1_2_padding = executable('strip_pkcs1_2_padding',
+    include_directories: core_inc,
+    sources: files('unittests' / 'strip_pkcs1_2_padding.c'),
+    link_with: [
+        libopensc,
+        libopensc_libraries
+    ],
+    dependencies: [
+        depcmocka,
+        libopensc_dependencies
+    ]
+)
+test('strip_pkcs1_2_padding', ut_strip_pkcs1_2_padding)
+
+if conf.get('ENABLE_OPENSSL')
+    ut_sm = executable('sm',
+        include_directories: core_inc,
+        sources: files('unittests' / 'sm.c'),
+        link_with: [
+            libopensc,
+            libopensc_libraries,
+            libsm
+        ],
+        dependencies: [
+            depcmocka,
+            libopensc_dependencies
+        ]
+    )
+    test('sm', ut_sm)
+endif
+
+if conf.get('ENABLE_ZLIB')
+    ut_compression = executable('compression',
+        include_directories: core_inc,
+        sources: files('unittests' / 'compression.c'),
+        link_with: [
+            libopensc,
+            libopensc_libraries
+        ],
+        dependencies: [
+            depcmocka,
+            libopensc_dependencies
+        ]
+    )
+    test('compression', ut_compression)
+endif
+
+#################################################
+# p11test
+#################################################
+if conf.get('ENABLE_OPENSSL')
+    executable('p11test',
+        include_directories: core_inc,
+        sources: files(
+            'p11test' / 'p11test.c',
+            'p11test' / 'p11test_case_common.c',
+            'p11test' / 'p11test_case_ec_derive.c',
+            'p11test' / 'p11test_case_ec_sign.c',
+            'p11test' / 'p11test_case_interface.c',
+            'p11test' / 'p11test_case_mechs.c',
+            'p11test' / 'p11test_case_multipart.c',
+            'p11test' / 'p11test_case_pss_oaep.c',
+            'p11test' / 'p11test_case_readonly.c',
+            'p11test' / 'p11test_case_secret.c',
+            'p11test' / 'p11test_case_usage.c',
+            'p11test' / 'p11test_case_wait.c',
+            'p11test' / 'p11test_case_wrap.c',
+            'p11test' / 'p11test_helpers.c',
+            'p11test' / 'p11test_loader.c',
+        ),
+        dependencies: [
+            depcmocka,
+            depopenssl
+        ]
+    )
+endif
+
+#################################################
+# Fuzzing tests
+#################################################
+fuzzer_main = get_option('fuzzing_libs') == ''? files('fuzzing' / 'fuzzer.c') : ''
+
+executable('fuzz_asn1_print',
+    include_directories: core_inc,
+    sources: [
+        files('fuzzing' / 'fuzz_asn1_print.c'),
+        fuzzer_main
+    ],
+    link_with: [
+        libopensc,
+        libopensc_libraries
+    ],
+    link_args: get_option('fuzzing_libs'),
+    dependencies: libopensc_dependencies
+)
```

Comment:
I think this repeats a a lot of logic for most of the fuzers. Is there a way to make some function/cycle over the fuzzers to avoid most of the copy-aste co

Reply:
Fixed locally.